### PR TITLE
[WFCORE-302] : Enable CORS

### PR DESCRIPTION
**Pull request by ehsavoie**
List of allowed-origins is parametrable and is filtering requests.
Security headers are set to enable authentication for all CORS requests.

Jira: https://issues.jboss.org/browse/WFCORE-302

### CODE REVIEW

**Comment by kabir** [comment]
This could do with a core model test to 'prove' the parsing of the new element works

**Comment by ehsavoie** [comment]
Test added

**Comment by bstansberry** [comment]
@darranl this needs your approval

File: server/src/main/resources/schema/wildfly-config_3_0.xsd [hunk 1]
```diff
@@ -2127,6 +2127,15 @@
                 </xs:choice>
                 <xs:attribute name="console-enabled" type="xs:boolean" use="optional" default="true"/>
                 <xs:attribute name="http-upgrade-enabled" type="xs:boolean" use="optional" default="false"/>
+                <xs:attribute name="allowed-origins" type="xs:string" use="optional" >
```
Comment (darranl) [inline server/src/main/resources/schema/wildfly-config_3_0.xsd]: I would prefer it if one of the list types was used for the attribute type which would make it a space separated list.  Any form of comma separated list requires reading of the documentation to understand how to format the content meaning tools can not do this automatically.

We already have a urlListType and stringListType in the schema that should be suitable for this.

**Comment by darranl** [comment]
Happy with the changes in principal, however as commented would prefer one of the list types is used instead of relying on documentation to let end users know how to format the attribute value.

The list types are already used elsewhere where users are configuring security for the management interfaces.

**Comment by ehsavoie** [comment]
Coma separated list string replaced by a space separated list.

**Pull request merged**